### PR TITLE
Add Changelog for 2.0.0 release

### PR DESCRIPTION
**Pull request by vigh-m**
**Description of changes:**

Updates the changelog with all the changes that went into the 2.0.0 release. There are some cross repo links to `bottlerocket-os/bottlerocket` since some of the work to enable 2.0.0 was done there.
**Terms of contribution:**

By submitting this pull request, I agree that this contribution is dual-licensed under the terms of both the Apache License, version 2.0, and the MIT license.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -68,3 +68,84 @@
...
+* Add libbpf ([#24])- Thanks ndbaker1!
+* Use Bottlerocket Settings SDK for settings models - ([#7])
+
+## OS Changes
+
+* Add libz dependency ([#31])
+* Update Rust dependencies ([#34])
+* Add Neuron ([#21])
+* Amazon SSM Agent dynamically linked binaries ([#22])
+* Add mdadm packages for software RAID ([#4035]) - Thanks tzneal!
```
Comment (KCSesh) [inline CHANGELOG.md:81]: Missing #4035 reference

Comment (vigh-m) [reply CHANGELOG.md:81]: Added!

**Comment by vigh-m** [comment]
Updated missing refs

**Review by KCSesh — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -68,3 +68,86 @@
 [#21]: https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/21
 [#22]: https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/22
 [#23]: https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/23
+
+# v2.0.0 (2024-06-14)
+
+## Kit Features
+
+* Move code to core kit ([#1])
+* Add libbpf ([#24])- Thanks ndbaker1!
```
Comment (bcressey) [inline CHANGELOG.md]: This was added for 2.2.0 and appears in the changelog already for that release.

File: CHANGELOG.md [hunk 3]
```diff
@@ -68,3 +68,86 @@
 [#21]: https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/21
 [#22]: https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/22
 [#23]: https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/23
+
+# v2.0.0 (2024-06-14)
```
Comment (bcressey) [inline CHANGELOG.md]: The release tag for 2.0.0 says June 20.
```suggestion
# v2.0.0 (2024-06-20)
```

File: CHANGELOG.md [hunk 4]
```diff
@@ -68,3 +68,86 @@
...
+* Move code to core kit ([#1])
+* Add libbpf ([#24])- Thanks ndbaker1!
+* Use Bottlerocket Settings SDK for settings models - ([#7])
+
+## OS Changes
+
+* Add libz dependency ([#31])
+* Update Rust dependencies ([#34])
+* Add Neuron ([#21])
+* Amazon SSM Agent dynamically linked binaries ([#22])
```
Comment (bcressey) [inline CHANGELOG.md]: #8 was the last PR merged for 2.0.0 per the [commit history](https://github.com/bottlerocket-os/bottlerocket-core-kit/commits/6939dd235ab4274e633b0022ffc812e00334648d/) - anything after that was a different release.

File: CHANGELOG.md (hunk 3 above)
Comment (vigh-m) [reply CHANGELOG.md]: Gotcha. I went off the merge date in #1

**Comment by vigh-m** [comment]
Updated following comments.

File: CHANGELOG.md [hunk 5]
```diff
@@ -68,3 +68,58 @@
 [#21]: https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/21
 [#22]: https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/22
 [#23]: https://github.com/bottlerocket-os/bottlerocket-core-kit/pull/23
+
+# v2.0.0 (2024-06-20)
+
+## Kit Features
+
+* Move code to core kit ([#1])
+* Use Bottlerocket Settings SDK for settings models - ([#7])
```
Comment (koooosh) [inline CHANGELOG.md]: nit: Remove `-` at end for consistency

File: CHANGELOG.md [hunk 6]
```diff
@@ -68,3 +68,58 @@
...
+
+## OS Changes
+
+* Add mdadm packages for software RAID ([#4035]) - Thanks tzneal!
+* Kernel Updates ([#4049],[#4039], [#4005], [#3972], [#3976])
```
Comment (koooosh) [inline CHANGELOG.md]: Should we list which kernels and to what versions like we do for newer releases?

**Review by koooosh — COMMENTED** [review]
Minor comments

File: CHANGELOG.md (hunk 6 above)
Comment (vigh-m) [reply CHANGELOG.md]: That makes sense. Updated

**Comment by vigh-m** [comment]
Addressed comments

**Review by koooosh — APPROVED** [review]

**Review by yeazelm — APPROVED** [review]

**Review by KCSesh — APPROVED** [review]

File: CHANGELOG.md [hunk 7]
```diff
@@ -68,3 +68,58 @@
...
+* Add mdadm packages for software RAID ([#4035]) - Thanks tzneal!
+* Update kernels: 5.10.217, 5.15.156, and 6.1.92([#4049],[#4039], [#4005], [#3972], [#3976])
+* Update containerd to 1.7.17 ([#4016])
+
+## Build Changes
+
+* Change pluto to act more like a settings generator ([#4032])
+* Update pluto for kits and Out of Tree Builds ([#3828])
+* Remove API Client dependency on the Settings model ([#3987])
+* Create Cloudformation settings extension ([#4010])
```
Comment (bcressey) [inline CHANGELOG.md]: nit: 
```suggestion
* Create CloudFormation settings extension ([#4010])
```

File: CHANGELOG.md [hunk 8]
```diff
@@ -68,3 +68,58 @@
...
+* Prepare os package for build system changes ([#4006])
+* Move to DNS settings extension ([#3980])
+* Move to OCI Hooks Settings Extension ([#3978])
+* Add Metrics Settings Extension ([#3963])
+* Move to PKI Settings Extension ([#3971])
+* Remove metadata migration ([#3958])
+* Remove version from makefile ([#4])
+* Improve cache behavior ([#6])
+* Twoliter updates ([#8])
+* Fix diff-kernel-config for core kit ([#32])
```
Comment (bcressey) [inline CHANGELOG.md]: this went out in 2.2.0
```suggestion
```

**Review by bcressey — COMMENTED** [review]
Looks good, just a couple of minor suggestions.

**Comment by vigh-m** [comment]
Updated commit message and changelog text

**Review by bcressey — APPROVED** [review]

**Pull request merged**